src/OrchardCore.Modules/OrchardCore.OpenId/Startup.cs

```diff
@@ -61,45 +65,97 @@ public override void ConfigureServices(IServiceCollection services)
+                           .ReplaceScopeManager(typeof(OpenIdScopeManager<>))
+                           .ReplaceTokenManager(typeof(OpenIdTokenManager<>));
+
+                    options.AddApplicationStore(typeof(OpenIdApplicationStore<>))
+                           .AddAuthorizationStore(typeof(OpenIdAuthorizationStore<>))
+                           .AddScopeStore(typeof(OpenIdScopeStore<>))
+                           .AddTokenStore(typeof(OpenIdTokenStore<>));
+
+                    options.SetDefaultApplicationEntity<OpenIdApplication>()
+                           .SetDefaultAuthorizationEntity<OpenIdAuthorization>()
+                           .SetDefaultScopeEntity<OpenIdScope>()
+                           .SetDefaultTokenEntity<OpenIdToken>();
+                });
+
+            services.TryAddScoped(provider => (IOpenIdApplicationManager) provider.GetRequiredService<IOpenIddictApplicationManager>());
+            services.TryAddScoped(provider => (IOpenIdAuthorizationManager) provider.GetRequiredService<IOpenIddictAuthorizationManager>());
+            services.TryAddScoped(provider => (IOpenIdScopeManager) provider.GetRequiredService<IOpenIddictScopeManager>());
+            services.TryAddScoped(provider => (IOpenIdTokenManager) provider.GetRequiredService<IOpenIddictTokenManager>());
+
+            services.AddSingleton<IIndexProvider, OpenIdApplicationIndexProvider>();
+            services.AddSingleton<IIndexProvider, OpenIdAuthorizationIndexProvider>();
+            services.AddSingleton<IIndexProvider, OpenIdScopeIndexProvider>();
+            services.AddSingleton<IIndexProvider, OpenIdTokenIndexProvider>();
+
+            services.AddScoped<IRoleRemovedEventHandler, OpenIdApplicationRoleRemovedEventHandler>();
+
+            services.AddScoped<IDataMigration, OpenIdMigrations>();
+        }
+    }
+
     [Feature(OpenIdConstants.Features.Server)]
     public class ServerStartup : StartupBase
     {
         public override void ConfigureServices(IServiceCollection services)
         {
             services.TryAddSingleton<IOpenIdServerService, OpenIdServerService>();
+            services.TryAddTransient<JwtBearerHandler>();
             services.AddScoped<IDisplayDriver<ISite>, OpenIdServerSettingsDisplayDriver>();
+            services.AddSingleton<IBackgroundTask, OpenIdBackgroundTask>();
+
+            services.AddRecipeExecutionStep<OpenIdServerSettingsStep>();
+            services.AddRecipeExecutionStep<OpenIdApplicationStep>();
 
-            services.TryAddScoped<IOpenIddictServerEventService, OpenIddictServerEventService>();
-            services.TryAddScoped<OpenIddictServerHandler>();
-            services.TryAddScoped<OpenIddictServerProvider>();
+            // Note: both the OpenIddict server and validation services are registered for the
+            // server feature as token validation may be required for the userinfo endpoint.
+            services.AddOpenIddict()
```

Comment:
Paradigm change here: so far, we used to manually register the OpenIddict server/validation services because couldn't use `services.AddOpenIddict().AddServer()/AddValidation()` as these extensions register the handlers in the authentication schemes options, which is something we can't do at this stage.

Thanks to an internal change in OpenIddict, it's now possible to use these extensions AND prevent the handlers from being registered here.

This comes with a great bonus: I'll be able to register new OpenIddict services without breaking the OrchardCore OpenID module once this is merged.